mchf-eclipse/drivers/audio/cw/cw_gen.c

```diff
@@ -226,7 +226,7 @@ static const float sm_table[CW_SMOOTH_TBL_SIZE] =
 
 void CwGen_SetSpeed()
 {
-    ps.dit_time         = 1650/ts.keyer_speed;      //100;
+    ps.dit_time         = 1920/ts.keyer_speed;      //100;
 }
```

Comment:
The problem is that the algorithm adds some ticks (a tick is 1s/1500)  to the pauses (you can see that in your measurements: the higher the WPM the longer the pause).  Still, with 1800 instead of 1920 we would have fairly correct dit + pause times (in sum):
-> ((1800/1920) * (208+224))/2 = 202.5 ms, 100.8 ms, ...   
 


  The problem is that the algorithm adds some ticks (a tick is 1s/1500)  to the pauses (you can see that in your measurements: the higher the WPM the longer the pause).  Still, with 1800 instead of 1920 we would have fairly correct dit + pause times (in sum):
-> ((1800/1920) * (208+224))/2 = 202.5 ms, 100.8 ms, 51.3 ,  26.5
I saw there is a 2 to 4 tick delay in the algorithm (which probably accounts for the 0.8 - 2.5ms additional time)  AND there is the ramp up and down of the edges which probably lead to seemingly shorter dits (and dahs).